### PR TITLE
Make scala plugin an optional dependency

### PR DESCRIPTION
**Pull request by wisechengyi**
This patch allows pants plugin to function without scala plugin. Reason being:
* some folks are only interested in java builds
* whenever there are scala plugin bugs, we will have to be impacted

### CODE REVIEW

**Comment by dotordogh** [comment]
Do you have examples of the bugs you're trying to avoid with this?

File: common/com/twitter/intellij/pants/util/PantsConstants.java [hunk 1]
```diff
@@ -25,6 +25,8 @@
   public static final String PANTS_INI = "pants.ini";
   public static final String PANTS_PEX = "pants.pex";
   public static final String PANTS_LIBRARY_NAME = PANTS_PEX;
+  public static final String ABSTRACT_TEST_RUN_CONFIGURATION_CLASS_NAME =
```
Comment (baroquebobcat) [inline common/com/twitter/intellij/pants/util/PantsConstants.java]: Since it's specific to Scala, should this constant have SCALA in it somewhere?

Comment (wisechengyi) [reply common/com/twitter/intellij/pants/util/PantsConstants.java]: Good idea!

**Comment by wisechengyi** [comment]
@dotordogh e.g. https://youtrack.jetbrains.com/oauth?state=%2Fissue%2FSCL-13783

**Comment by wisechengyi** [comment]
@baroquebobcat good question on the test. It probably needs a separate run script that does the test without scala plugin on the classpath.

**Pull request merged**